### PR TITLE
[RFR] Migrate ra-ui-materialui to TypeScript (Part 2: fields)

### PR DESCRIPTION
**Pull request by Kmaschta**
Wish me luck.

## TODO
- [x] Migrate fields components to TS
- [x] Use `Identifier` instead of string everywhere an `id` is used
- [X] Use the `withTranslate` from Gildas' PR

### CODE REVIEW

File: packages/ra-ui-materialui/src/field/BooleanField.tsx [hunk 1]
```diff
@@ -55,7 +66,7 @@ export const BooleanField = ({
         return (
             <Typography
                 component="span"
-                body1="body1"
```
Comment (Kmaschta) [inline packages/ra-ui-materialui/src/field/BooleanField.tsx]: This was a common bug. Bad copy/paste

**Comment by djhi** [comment]
I'll work on the remaining fields in a separate PR

File: packages/ra-ui-materialui/src/field/types.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    label?: string;
+    sortable?: boolean;
+    className?: string;
+    cellClassName?: string;
+    headerClassName?: string;
+    textAlign?: TextAlign;
+}
+
+// Props injected by react-admin
+export interface InjectedFieldProps {
```
Comment (djhi) [inline packages/ra-ui-materialui/src/field/types.ts]: I think this should be exported by `ra-core`

**Comment by djhi** [comment]
I'm going to revert the changes on all the Reference fields components. I believe they should leverage types declared but not yet exported from `ra-core`.

File: packages/ra-ui-materialui/src/detail/SimpleShowLayout.spec.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ describe('<SimpleShowLayout />', () => {
                 <TextField source="bar" />
             </SimpleShowLayout>
         );
-        const inputs = wrapper.find('pure(TextField)');
+        const inputs = wrapper.find('EnhancedTextField');
```
Comment (djhi) [inline packages/ra-ui-materialui/src/detail/SimpleShowLayout.spec.js]: By setting the displayName on the `EnhancedTextField`, we can avoid testing implementation details here

**Comment by Kmaschta** [comment]
There are still some `Component.propTypes` in this PR, are they still needed?

**Comment by djhi** [comment]
Actually I added them back. Yes they are necessary for run time check for people not using TS

**Comment by fzaninotto** [comment]
needs rebase following changes on BooleanFields

**Pull request merged**